internal/engine/synclet_build_and_deployer.go

```diff
@@ -124,3 +134,63 @@ func (sbd *SyncletBuildAndDeployer) updateViaSynclet(ctx context.Context,
 	resultSet[image.ID()] = res
 	return resultSet, nil
 }
+
+func (sbd *SyncletBuildAndDeployer) updateViaSynclet(ctx context.Context,
+	podID k8s.PodID, namespace k8s.Namespace, containerID container.ID,
+	archive *bytes.Buffer, filesToDelete []string, cmds []model.Cmd, hotReload bool) error {
+	sCli, err := sbd.sm.ClientForPod(ctx, podID, namespace)
+	if err != nil {
+		return err
+	}
+
+	err = sCli.UpdateContainer(ctx, containerID, archive.Bytes(), filesToDelete, cmds, hotReload)
+	if err != nil && build.IsUserBuildFailure(err) {
+		return WrapDontFallBackError(err)
+	}
+	return err
+}
+
+func (sbd *SyncletBuildAndDeployer) updateViaExec(ctx context.Context,
+	podID k8s.PodID, namespace k8s.Namespace, container container.Name,
+	archive *bytes.Buffer, archivePaths []string, filesToDelete []string, cmds []model.Cmd, hotReload bool) error {
+	l := logger.Get(ctx)
+	w := l.Writer(logger.InfoLvl)
+
+	if len(filesToDelete) > 0 {
+		filesToShow := filesToDelete
+		if len(filesToShow) > 5 {
+			filesToShow = append([]string(nil), filesToDelete[0:5]...)
+			filesToShow = append(filesToShow, "...")
+		}
+
+		l.Infof("removing %v files %v", len(filesToDelete), filesToShow)
```

Comment:
the logging in this file seems like it might be a bit much, and more than what we do for other builders - do you have feelings on this?

Reply:
The in-container building is weirder, and so I think worth being more explicit, so that people can debug when things go wrong instead of blaming us.

Reply:
Yeah considering this is an alpha feature I feel OK leaving this in. We can always remove it later.